docker-compose.yml.erb

```diff
@@ -7,6 +7,7 @@ services:
         ASSETS_S3_BUCKET: '<%= config.asset_bucket %>'
         DEVELOPMENT_MODE: 'true'
         RUN_BUILD: 'false'
+    image: quay.io/organisation/<%= config.name %>:dev
```

Comment:
note this will cause the `docker-compose down -v -rmi local` to not delete this.

I'm a bit confused what scenario it solves, I suppose its if due to host file changes that's what would trigger a different COPY?

Reply:
To build a docker image, the context being built needs to be compressed and sent over HTTP(S) to the docker daemon, which then extracts it.

We've seen that for large contexts, this can take up to 10 minutes, which is then repeated for the setup container (where the setup container is separate from web to ensure the stack stays up when other things like varnish are used).

Once the context is uploaded it's generally quite fast (using cache) but there have been occasions where it repeated the same build steps as web from fresh.

Reply:
ok, how about avoiding a setup container for dev then, just having a setup docker-compose exec somehow or in parallel to the web do_start_supervisor?